kairos-server/src/main.rs

```diff
@@ -1,21 +1,44 @@
-use std::net::SocketAddr;
+use lazy_static::lazy_static;
 
-use dotenvy::dotenv;
-use kairos_server::{config::ServerConfig, state::BatchState};
+use kairos_server::{config::Settings, state::BatchState};
+
+// Globally accessible config
+lazy_static! {
+    static ref CONFIG: Settings = Settings::new();
+}
 
 #[tokio::main]
 async fn main() {
-    tracing_subscriber::fmt::init();
-    dotenv().ok();
-
-    let config = ServerConfig::from_env()
-        .unwrap_or_else(|e| panic!("Failed to parse server config from environment: {}", e));
+    CONFIG.initialize_logger();
 
     let app = kairos_server::app_router(BatchState::new());
 
-    let axum_addr = SocketAddr::from(([127, 0, 0, 1], config.port));
+    let axum_addr = CONFIG.socket_address();
 
-    tracing::info!("starting http server on `{}`", axum_addr);
+    tracing::info!("Starting HTTP server on `{}`", axum_addr);
     let listener = tokio::net::TcpListener::bind(axum_addr).await.unwrap();
-    axum::serve(listener, app).await.unwrap();
+    axum::serve(listener, app)
+        .with_graceful_shutdown(shutdown_signal())
+        .await
+        .unwrap();
+}
+
+async fn shutdown_signal() {
+    let ctrl_c = async {
+        tokio::signal::ctrl_c()
+            .await
+            .expect("Failed to install Ctrl+C handler");
+    };
+
+    let terminate = async {
```

Comment:
I'm pretty sure this is what tokio already does for us when we send SIGTERM. There is no need to implement this ourselves, especially given that we don't need to handle any special conditions when we terminate

Reply:
![image](https://github.com/cspr-rad/kairos/assets/58150534/40b91af9-7deb-4e37-a4c2-aad3be8bf294)
Different behavior is observable. Will probably be useful to wait for database connections to close before shutdown in the future as well.

Reply:
The crate I'm using for loading the config is super flexible, I could add CLI options as well if that's something you'd like to see. I would really like to leave the toml config in place however, because it's flexible. For example, if I or anyone else wants to provide a list of nodes for the deposit listener to connect to, or more likely a list of nodes we know are trustworthy or low latency, toml is pretty practical. I can go to the config file and simply type the following:

```
[[nodes]]
address = "192.168.1.1"
port = 11101

[[nodes]]
address = "12.168.1.1"
port = 11101
```

But dotenv out of the box doesn't support complex structures and while a cli could definitely do it, it's a little cumbersome.

Reply:
The issue is it's duplicate code that does not currently add functionality.
In the future it might make sense, but I don't anticipate having a complex set of args.

In this case nodes can be a CSV String `address:port,*`.
If we have a super complex set of structured args in the future we can revisit using toml.